### DISCUSSION TITLE
Is it possible to pass extensions to the server?

### DISCUSSION THREAD

**Discussion by pdong** [urql-graphql/urql#1847] [Q&A] [answered]
I'm investigating migrating from Apollo to urql and I'm trying to figure out if it is possible to send `extensions` to the server as part of the request.  Apollo has a configuration `includeExtensions` https://www.apollographql.com/docs/react/networking/advanced-http-networking/#constructor-options which you can use then add extensions via `operation.extensions`.

I tried looking at fetchOptions where you can return `RequestInit` to set headers but setting extensions in the body here doesn't seem to do anything e.g.

```
fetchOptions: () => {
    return {
      body: {
        extensions: {
          someToken: "token",
        },
      },
    };
  },
```

I was hoping to be able to use `authExchange` to add an extension with a token it as the API I'm using expects a token in extensions for authorization.  I'm assuming I'd use `addAuthToOperation` but I'm running into the same issue where `OperationContext` has fetchOptions but no way to modify the body.

 It looks like the default fetchExchange always defaults `extensions` to undefined when creating the body https://github.com/FormidableLabs/urql/blob/ef4b5b002d31e17d84fb41e5feb24d2200c9656f/packages/core/src/internal/fetchOptions.ts#L17-L25

I'm guessing what I'm trying to do is currently not possible without a custom fetchExchange, right? (Would that even be the correct approach to solve this?) Hopefully I'm missing something obvious 🤞 Thanks.

**Comment by JoviDeCroock** [comment, marked as answer]
Currently we don't support this as most servers also don't really support it, I think in general it's safer to pass tokens like this:

```js
fetchOptions: () => {
    return {
      headers: {
          authorization: "token",
      },
    };
  },
```

Any specific reason you prefer using `extensions`? Generally from the graphql-spec we can see that `extensions` is meant to extend the spec.

I guess if we were to add this it would be part of the `OperationContext` rather than something in `fetchOptions`,....
